buildroot/share/PlatformIO/scripts/common-features-dependencies.h

```diff
@@ -48,3 +48,4 @@
 #include "../../../../Marlin/Configuration_adv.h"
 
 #include "../../../../Marlin/src/inc/Conditionals_adv.h"
+#include "../../../../Marlin/src/inc/Conditionals_post.h"
```

Comment:
Some items in this file depend on values that can only be obtained from `HAL.h` and `pins.h`, which both depend on having the environment values provided by the active target.

Reply:
I added `Conditionals_post.h` to use HAS_TMC_SW_SERIAL, to link it to with SoftwareSerial... but after I discovered that SoftwareSerial is very very trick, so I gave up.
Its really aren't used right now. I just keeped it because I thought that it could have more defines we need. 

We can remove, if you think its more safe.